src/lib/pubkey/pbes2/pbes2.cpp

```diff
@@ -85,15 +87,15 @@ secure_vector<uint8_t> derive_key(std::string_view passphrase,
 
       auto pwdhash_fam = PasswordHashFamily::create_or_throw("Scrypt");
       auto pwdhash = pwdhash_fam->from_params(N, r, p);
-      pwdhash->derive_key(derived_key.data(), derived_key.size(),
-                          passphrase.data(), passphrase.size(),
-                          salt.data(), salt.size());
+      pwdhash->hash(derived_key, passphrase, salt);
 
       return derived_key;
       }
    else
-      throw Decoding_Error("PBE-PKCS5 v2.0: Unknown KDF algorithm " +
-                           kdf_algo.oid().to_string());
+      {
+      throw Decoding_Error(fmt("PBE-PKCS5 v2.0: Unknown KDF algorithm {}",
+                                kdf_algo.oid().to_string()));
```

Comment:
As a side note: When we eventually move to `std::format` these `to_string()` gymnastics should be handled centrally with our custom types IMO. Didn't look into it in detail, but I'd be surprised if one could not define this generically for user-defined types in `std::format`.

For now, I think, this is totally sufficient.

Reply:
I haven't looked into in detail either (seems somewhat academic without a compiler to test anything with) but it appears [std::formatter](https://en.cppreference.com/w/cpp/utility/format/formatter) will let us add custom flags support and other fancy stuff.

For the short term I added an operator<< for OID, which is sufficient to avoid the explicit `to_string` calls, since our `fmt` will happily eat anything it can put into an `ostream`